pandas/tests/indexes/test_datetimelike.py

```diff
@@ -732,6 +741,13 @@ def test_fillna_datetime64(self):
                            dtype=object)
             self.assert_index_equal(idx.fillna('x'), exp)
 
+    def test_difference_freq(self):
+        # GH14323. difference of DatetimeIndex does not preserve frequency,
+        # so a differencing operation should not retain the freq field of the
+        # original index.
+        i = self.create_index()
+        self.assertIsNone(i.difference(i[1:3]).freq)
```

Comment:
use assert_index_equal against a constructed expected index (same for ones below)

for dti and tdi do this in the superclass override for pi